### ISSUE TITLE
Selected Network Interface won't come into use unless FrostWire is restarted.

### ISSUE THREAD

**Issue by gubatron** [frostwire/frostwire-desktop#393]
An advanced user that connects via a VPN Network interface says as of FrostWire 6, selecting the network interface does not take into effect unless the client is restarted.
(Probably the user interface said that before, but the setting did take effect as all networking was done directly in Java with the old Azureus library, we'll have to see how the network interface binding happens now that we're using libtorrent)

According to him this was not the case with FrostWire 5.7.x.

I'll proceed to perform these tests to replicate the issue, hopefully frostwire-libtorrent will bind to the new network interface as specified by the user.

**Comment by aldenml** [comment]
I remember Vuze some kink of thread/timer checking for a stale network and downloads. I don't see any of this in libtorrent.

**Comment by muckachina** [comment]
mmm I haven't test under VPN yet, I'll buy a subscription because there are several reports having issues running FW under VPN, I know Kademlia has checked some of them, but now that you also "highly recommend" the use of VPN when installing FrostWire is important to make sure the app will run correctly.

**Comment by gubatron** [comment]
I believe this should be fixed now.
Testing this will probably require using something like Wireshark, to make sure FrostWire is indeed using the selected network interface.

**Comment by gubatron** [comment]
Please build before submitting PRs in the meantime.

You can build from the command line using:

`./gradlew assembleDebug`

**Comment by gubatron** [comment]
@muckachina please test this to see if we can close this issue.
See http://wireshark.org

**Comment by muckachina** [comment]
I'll have a look, I've to get the VPN first, then analyze the network with wireshark.

**Comment by muckachina** [comment]
Ok, I'm not pretty sure if this is what I had to check, but I installed Wireshark --> connected a Windows PC via cable and WiFi and then --> after selecting manually the Network Interface --> I captured the packets using **Wireshark** to make sure FrostWire was using the specified network, so this is the result:

- My selected network showed transference of packets after being selected on FrostWire but all of them appears at the statistics as **"queued for transmission"**

- After searching and downloading files the other two connections different to FW network interface shows packets transmission (transmitted, Received, etc and none queued) so **I assume FrostWire is not using the specified address for transmission**

![screen shot 2015-02-19 at 8 28 21 pm](https://cloud.githubusercontent.com/assets/1534723/6280038/ec6afef4-b875-11e4-814a-f79de11a794f.png)

Did you test this on Mac? I'll ask Carlos to bring me his ethernet-usb connector for Mac from the office so I can test on Mac tomorrow.

**Comment by gubatron** [comment]
hmm, I'm not sure testing this on VirtualBox is the right approach, as VirtualBox really has no choice but to use the VBox adapter...

this should be tested on a native OS, I'll test on MacOSX.

In whatever case, test both, right after applying the option, and also after restarting FrostWire to see how it behaves. The packets on that window up there should be good enough to see what iface is being used.

**Comment by muckachina** [comment]
I tried to run Wireshark on Mac under X11 app with no success, there is library missing and I couldn't run it, so I tried with nettop which is the command on Mac to monitor traffic and connection and this is what I have:

--> I connected my Mac via ethernet cable and FrostWire was running using WiFi connection
--> Changed the Network Interface on *Tools > Options > Advanced > Network Interface* and choose "en3" instead of "en0" that showed to have the main activity before connecting the ethernet cable
--> Launched again the command **nettop** and checked the interface "en3" was being used and transferring packets, however I couldn't check if it was FW the main app triggering these transfers.

I also tested with another network monitor tool named **iStat** for Mac and did the same, however this time it was more evident the use of the ethernet connection every time I downloaded using FrostWire so I can assume FrostWire is using the ethernet connection specified within the advanced options.

![screen shot 2015-02-20 at 3 29 14 pm](https://cloud.githubusercontent.com/assets/1534723/6308902/245cacfe-b916-11e4-8378-694ed4230032.png)

And finally, after all these, I read this piece of info: *"Mac will always use the network connections in order that's listed in your "Network Connections" profile. For example, if Ethernet is on top of Wireless, Ethernet will take over"*
So, I also assume no matter what I change on FrostWire, on Mac the network interface used by default will be determined by the Mac configuration network. 

o.O
